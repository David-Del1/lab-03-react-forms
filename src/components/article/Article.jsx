import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ article }) => (
  <div>
    <h1>{article.title}</h1>
    <p>{article.author}</p>
    <p> {article.description}</p>
  </div>
);

Article.propTypes = {
  article: PropTypes.object.isRequired
};

export default Article;
