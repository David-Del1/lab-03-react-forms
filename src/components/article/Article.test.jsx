import React from 'react';
import { render } from '@testing-library/react';
import Article from './Article';

describe('Article component', () => {
  it('renders articles', () => {
    const article = {
      title: 'bitcoin',
      author: 'David Delgadillo',
      description: 'nice'
    };

    const { asFragment } = render(<Article article={article}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
