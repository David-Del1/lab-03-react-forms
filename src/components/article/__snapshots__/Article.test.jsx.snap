// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders articles 1`] = `
<DocumentFragment>
  <div>
    <h1>
      bitcoin
    </h1>
    <p>
      David Delgadillo
    </p>
    <p>
       nice
    </p>
  </div>
</DocumentFragment>
`;
